### PR TITLE
[WIP] Bump to scala 2.13/chisel 3.5.5/latest rocketchip

### PR DESCRIPTION
**Pull request by jerryz123**

### CODE REVIEW

File: src/main/scala/Serdes.scala [hunk 1]
```diff
@@ -555,7 +556,8 @@ class TLDesser(w: Int, params: Seq[TLClientParameters], hasCorruptDenied: Boolea
   val node = TLClientNode(params.map(client =>
       TLMasterPortParameters.v1(Seq(client))))
 
-  lazy val module = new LazyModuleImp(this) {
+  lazy val module = new Impl
+  class Impl extends LazyModuleImp(this) {
```
Comment (abejgonzalez) [inline src/main/scala/Serdes.scala:560]: Why is this needed/done?

**Review by abejgonzalez — APPROVED** [review]

File: src/main/scala/TraceIO.scala [hunk 2]
```diff
@@ -13,45 +13,17 @@ import freechips.rocketchip.tile.{BaseTile}
 import freechips.rocketchip.diplomacy.{BundleBridgeSource, BundleBroadcast, BundleBridgeNexusNode}
 
 //***************************************************************************
-// Extended Trace Instruction Utilities:
-// Used to map TracedInstructions to their Extended version and
```
Comment (abejgonzalez) [inline src/main/scala/TraceIO.scala:17]: Why is this getting removed?

Comment (jerryz123) [reply src/main/scala/TraceIO.scala:17]: I upstreamed this to rocket-chip

File: src/main/scala/Serdes.scala (hunk 1 above)
Comment (jerryz123) [reply src/main/scala/Serdes.scala:560]: Result of change in scala2.13. This change was applied via scalafix, and all rocket-chip dependencies on 2.13 had this change applied

**Review by abejgonzalez — APPROVED** [review]
